src/actor-context.lisp

```diff
@@ -107,22 +107,37 @@ An `act:actor` contains an `actor-context`."
     context))
 
 (defmethod actor-of ((context actor-context)
-                     &key receive
-                       (init nil) (destroy nil)
-                       (dispatcher :shared) (state nil)
-                       (type 'act:actor) (name nil)
-                       (other-args nil))
+                     &rest rest
+                     &key
+                     receive
+                     (init nil)
+                     (destroy nil)
+                     (dispatcher :shared)
+                     (state nil)
+                     (type 'act:actor)
+                     (name nil)
+                     (queue-size nil)
+                     &allow-other-keys)
   "See `ac:actor-of`."
   (check-type receive function "a function!")
+  (alexandria:remove-from-plistf rest
```

Comment:
Is this necessary? IMO it doesn't look great and adds an additional burden to strip keys here.

Reply:
If these keys are not removed from the REST, they will be passed along to the act:make-actor and will cause error like:

``
Invalid initialization argument:
  :DISPATCHER"


 0: (SB-PCL::INITARG-ERROR #<STANDARD-CLASS SENTO.DISPATCHER:DISPATCH-WORKER> (:DISPATCHER))
 1: ((:METHOD MAKE-INSTANCE (CLASS)) #<STANDARD-CLASS SENTO.DISPATCHER:DISPATCH-WORKER> :RECEIVE #<FUNCTION SENTO.DISPATCHER::RECEIVE> :INIT N$
 2: ((:METHOD MAKE-ACTOR (T)) #<FUNCTION SENTO.DISPATCHER::RECEIVE> :INIT NIL :DESTROY NIL :STATE NIL :NAME "dispatch(SHARED)-worker-1" :RECEI$
      Locals:
        #:.DEFAULTING-TEMP. = SENTO.DISPATCHER:DISPATCH-WORKER
        SENTO.ACTOR::ALL-ARGS = (:INIT NIL :DESTROY NIL :STATE NIL :NAME "dispatch(SHARED)-worker-1" :RECEIVE #<FUNCTION SENTO.DISPATCHER::REC$
        SENTO.ACTOR::RECEIVE = #<FUNCTION SENTO.DISPATCHER::RECEIVE>
        TYPE = SENTO.DISPATCHER:DISPATCH-WORKER
 3: ((SB-PCL::EMF MAKE-ACTOR) #<unused argument> #<unused argument> #<FUNCTION SENTO.DISPATCHER::RECEIVE> :INIT NIL :DESTROY NIL :STATE NIL :T$
 4: ((LAMBDA NIL :IN SENTO.ACTOR-CONTEXT:ACTOR-OF))
 5: (SENTO.ACTOR-CONTEXT::%CREATE-ACTOR #<SENTO.ACTOR-CONTEXT:ACTOR-CONTEXT {1006010683}> #<FUNCTION (LAMBDA NIL :IN SENTO.ACTOR-CONTEXT:ACTOR$
      Locals:
        CONTEXT = #<SENTO.ACTOR-CONTEXT:ACTOR-CONTEXT {1006010683}>
        CREATE-FUN = #<FUNCTION (LAMBDA () :IN SENTO.ACTOR-CONTEXT:ACTOR-OF) {10061575BB}>
        DISPATCHER-ID = :PINNED
        QUEUE-SIZE = NIL
 6: (SENTO.ACTOR-CONTEXT::%ACTOR-OF #<SENTO.ACTOR-CONTEXT:ACTOR-CONTEXT {1006010683}> #<FUNCTION (LAMBDA NIL :IN SENTO.ACTOR-CONTEXT:ACTOR-OF)$
 7: ((:METHOD SENTO.ACTOR-CONTEXT:ACTOR-OF (SENTO.ACTOR-CONTEXT:ACTOR-CONTEXT)) #<SENTO.ACTOR-CONTEXT:ACTOR-CONTEXT {1006010683}> :RECEIVE #<F$
 8: ((SB-PCL::EMF SENTO.ACTOR-CONTEXT:ACTOR-OF) #<unused argument> #<unused argument> #<SENTO.ACTOR-CONTEXT:ACTOR-CONTEXT {1006010683}> :RECEI$
 9: (SENTO.DISPATCHER:MAKE-DISPATCHER-WORKER 1 #<SENTO.ACTOR-CONTEXT:ACTOR-CONTEXT {1006010683}> :SHARED)
10: ((:METHOD INITIALIZE-INSTANCE :AFTER (SENTO.DISPATCHER:SHARED-DISPATCHER)) #<SENTO.DISPATCHER:SHARED-DISPATCHER ident: SHARED, workers: 0,$
11: ((LAMBDA (SB-PCL::|.P0.| SB-PCL::|.P1.| SB-PCL::|.P2.| SB-PCL::|.P3.|)) #<unavailable argument> #<unavailable argument> #<unavailable argu$
12: (SENTO.DISPATCHER:MAKE-DISPATCHER #<SENTO.ACTOR-CONTEXT:ACTOR-CONTEXT {1006010683}> :SHARED :WORKERS 4 :STRATEGY :RANDOM)
13: (SENTO.ACTOR-SYSTEM::%REGISTER-DISPATCHERS #<SENTO.ACTOR-SYSTEM:ACTOR-SYSTEM config: (SCHEDULER (ENABLED TRUE RESOLUTION 100 MAX-SIZE 500)$
14: (SENTO.ACTOR-SYSTEM:MAKE-ACTOR-SYSTEM (:DISPATCHERS (:SHARED (:WORKERS 4 :STRATEGY :RANDOM))))
15: ((LABELS %TEST-COUNTER-MP-UNBOUNDED :IN "/home/art/.cache/common-lisp/sbcl-2.4.4-linux-x64/home/art/projects/cl-gserver/tests/actor-mp-tes$
16: ((LABELS IT.BESE.FIVEAM::RUN-IT :IN IT.BESE.FIVEAM::RUN-TEST-LAMBDA))
```

This error is generated because actor class does not have `:DISPATCHER` initarg.

I've found that in this case we can replace lengthy:

```
(alexandria:remove-from-plistf rest
                                 :queue-size
                                 :dispatcher
                                 :init
                                 :destroy
                                 :state
                                 :type
                                 :name)
```

With only:

```
(alexandria:remove-from-plistf rest
                                 :queue-size
                                 :dispatcher)
```

or we can remove `alexandria:remove-from-plistf` completely but pass `:allow-other-keys t` to the `make-instance`. But this way we will turn off check of keyword argument and library user might be surprised when hi mistype an argument name and actor's slot will not be filled with the value.

I'd go with a smaller number of excluded argument names but not turn off names checking.

Reply:
OK